src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -751,6 +741,117 @@ private void ReturnReadBufferIfEmpty()
                 _decryptedBytesCount = 0;
                 _decryptedBytesOffset = 0;
             }
+            else if (_decryptedBytesCount == 0)
+            {
+                _decryptedBytesOffset = 0;
+            }
+        }
+
+        private bool haveFullTlsFrame(ref int frameSize)
```

Comment:
`have` => `Have`, similar below

Reply:
Also `frameSize` should be `out` right?

Reply:
Or possibly return a tuple (bool hasFrame, int frameSize). Or just return an int and use -1 to mean not having a full frame.